### ISSUE TITLE
Put some examples in the documentation

### ISSUE THREAD

**Issue by robbporto** [acdlite/recompose#731]
It would be very nice to put some "real world" examples when explaining each method. Right now we mostly have only the signature of the method.

**Comment by suzdalnitski** [comment]
I agree that more examples should be added. We're using recompose heavily at work, and I have even written an article: [Mastering React Functional Components with Recompose](https://medium.com/@ilyasz/mastering-react-functional-components-with-recompose-d4dd6ac98834)

If the examples in the article look good, then I could add them to the documentation, perhaps add a whole new **Examples** section.

**Comment by markusgattol** [comment]
To me the real value add when we stay with functions rather than classes is that testing becomes a lot easier and doing TDD is something people might actually do which in turn is the only thing that gets you scalable codebases when you grow a business.

Therefore, I'd like to see info in the docs that reflects on that as well, mention it and give examples using e.g. Jest.
